### DISCUSSION TITLE
Executable binary/CLI that wraps NextJS and parses local `.mdx` files

### DISCUSSION THREAD

**Discussion by thiskevinwang** [vercel/next.js#28722] [Help] [answered]
# Goal

_As a non-nextjs / non-front-end developer, I want to run a script like `my-custom-next-bin --someflag=foo ./path/to/docs`_

_It should parse my local repository, which will contain some `.mdx` files, but no trace of NextJS code._

_It should spin up a local NextJS app @ port, and render pages based on the previous `.mdx` files_

(This NextJS app would be maintained in a separate repository)

---

At first `docker` seems like an option — I'd pass in arbitrary file directories to _**this**_ custom image — but seems a bit clunky.

How would one achieve this? 🤔 

Thanks in advance for any tips! 
Let me know if this needs some clarification!

**Comment by adogg43** [comment]
you should clean this website for users having unqualifying standards.
pictures are being exploded and

**Comment by thiskevinwang** [comment, marked as answer]
Figured out a bare-bones option w/ `docker`

1. Build docker image
    - the nextjs code would need to know how to handle a certain directory — ex. `./content`
2. Share that docker image
    - either a `.tar.gz` via `docker save ...`
    - or pullable docker image from _some_ container registry, via `docker push` & `docker pull`
3. Run the container
    - `docker run --rm -p 3000:3000 --name=my_container username/image_name`
4. Copy local files
    - `docker cp content/. my_container:/app/content`
5. Maybe merge `4` and `5` into a `docker-compose.yaml` file, and eventually have a single command to run.

It's _clunky_, but it works!
